### ISSUE TITLE
False positive no-member: Module 'colorful' has no 'red' member

### ISSUE THREAD

**Issue by L3viathan** [PyCQA/pylint#3914]
### Steps to reproduce

After installing [colorful](https://pypi.org/project/colorful/), call pylint on the following code:

```
import colorful as cf

print(cf.red("Hello World!"))
```

### Current behavior

```
example.py:3: [E1101(no-member), ] Module 'colorful' has no 'red' member
```

### Expected behavior

No error, as colorful dynamically generates its members

### pylint --version output

```
pylint 2.6.0
astroid 2.4.2
Python 3.6.9 (default, Jul 17 2020, 12:50:27)
[GCC 8.4.0]
```

----

I'm aware I can circumvent this by adding `colorful.*,cf.*` to the `generated-members` setting, but that matches by name, which means _anything_ imported as e.g. `cf` will be ignored with regards to the `no-member` error.

I would expect a `# pylint: disable=no-member` on the import line to disable the check in this module only for the imported module, but that is not the case.

**Comment by hippo91** [comment]
@L3viathan thanks for the report. It appears that, as you mentioned it, `colorful` generates parts of its member dynamically. Thus it is not possible for `pylint`, that is a static tool, to understand this.
However you can write an `astroid` brain. `astroid` is the inference engine used by `pylint` and a brain is just a piece of python code that mocks the behavior of the desired module, in this case `colorful`.

**Comment by PCManticore** [comment]
Hey folks, thanks for the discussion. As already indicated, this can be "fixed" either by using `--generated-members` or `--ignored-modules` or by writing an astroid brain plugin that you can use in your project. I don't think we should add it to the  upstream `astroid` project though, as we really need to choose between more popular libraries and we can't maintain all the brains that we need.
